go/client/chat_resolver.go

```diff
@@ -45,41 +45,71 @@ type chatConversationResolvingBehavior struct {
+	TeamsClient keybase1.TeamsInterface
+}
+
+func newChatConversationResolver(g *libkb.GlobalContext) (c *chatConversationResolver, err error) {
+	c = new(chatConversationResolver)
+	if c.ChatClient, err = GetChatLocalClient(g); err != nil {
+		return c, err
+	}
+	if c.TlfClient, err = GetTlfClient(g); err != nil {
+		return c, err
+	}
+	if c.TeamsClient, err = GetTeamsClient(g); err != nil {
+		return c, err
+	}
+	c.G = g
+	return c, nil
 }
 
 // completeAndCanonicalizeTLFName completes tlfName and canonicalizes it if
 // necessary. The new TLF name is stored to req.ctx.canonicalizedTlfName.
 // len(tlfName) must > 0
 func (r *chatConversationResolver) completeAndCanonicalizeTLFName(ctx context.Context, tlfName string, req chatConversationResolvingRequest) error {
 
-	query := keybase1.TLFQuery{
-		TlfName:          tlfName,
-		IdentifyBehavior: keybase1.TLFIdentifyBehavior_CHAT_CLI,
-	}
-	var cname keybase1.CanonicalTLFNameAndIDWithBreaks
-	var err error
-	var visout string
-	if req.Visibility == keybase1.TLFVisibility_PUBLIC {
-		visout = "public"
-		cname, err = r.TlfClient.PublicCanonicalTLFNameAndID(ctx, query)
-	} else {
-		visout = "private"
-		cname, err = r.TlfClient.CompleteAndCanonicalizePrivateTlfName(ctx, query)
-	}
-	if err != nil {
-		// When a recipient's proofs are failing, this is the error a CLI user
-		// will see. It needs to be human readable.
-		return fmt.Errorf("failed to open chat conversation: %v", err)
-	}
-	if string(cname.CanonicalName) != tlfName {
-		// If we auto-complete TLF name, we should let users know.
-		// TODO: don't spam user here if it's just re-ordering
-		r.G.UI.GetTerminalUI().Printf("Using %s conversation %s.\n", visout, cname.CanonicalName)
+	switch req.MembersType {
+	case chat1.ConversationMembersType_KBFS:
+		query := keybase1.TLFQuery{
+			TlfName:          tlfName,
+			IdentifyBehavior: keybase1.TLFIdentifyBehavior_CHAT_CLI,
+		}
+		var cname keybase1.CanonicalTLFNameAndIDWithBreaks
+		var err error
+		var visout string
+		if req.Visibility == keybase1.TLFVisibility_PUBLIC {
+			visout = "public"
+			cname, err = r.TlfClient.PublicCanonicalTLFNameAndID(ctx, query)
+		} else {
+			visout = "private"
+			cname, err = r.TlfClient.CompleteAndCanonicalizePrivateTlfName(ctx, query)
+		}
+		if err != nil {
+			// When a recipient's proofs are failing, this is the error a CLI user
+			// will see. It needs to be human readable.
+			return fmt.Errorf("failed to open chat conversation: %v", err)
+		}
+		if string(cname.CanonicalName) != tlfName {
+			// If we auto-complete TLF name, we should let users know.
+			// TODO: don't spam user here if it's just re-ordering
+			r.G.UI.GetTerminalUI().Printf("Using %s conversation %s.\n", visout, cname.CanonicalName)
+		}
+		req.ctx.canonicalizedTlfName = string(cname.CanonicalName)
+	case chat1.ConversationMembersType_IMPTEAM:
+		impRes, err := r.TeamsClient.LookupOrCreateImplicitTeam(ctx, keybase1.LookupOrCreateImplicitTeamArg{
```

Comment:
You want `OrCreate` here?

Reply:
Yes, if I send to `mikem,mlsteele` and it doesn't exist yet, I want it to be created. This is the same behavior as KBFS.